### ISSUE TITLE
`RUBY_ENGINE_VERSION` and `JRUBY_VERSION` return version with suffix `SNAPSHOT` instead of `dev` for a dev version

### ISSUE THREAD

**Issue by bogdanvlviv** [jruby/jruby#5144]
### Environment

JRuby: jruby-9.2.0.0-dev
rbenv.

Recently we patched Rails in order to create the `.ruby-version` compatible with MRI/JRuby by default. See issue https://github.com/rails/rails/issues/32639.
We use `RUBY_ENGINE_VERSION`(on `master` branch, https://github.com/rails/rails/commit/dc8a66aa47318b3a6aaf4014b9f118c51edb67e8) and `JRUBY_VERSION`(on `5-2-stable` branch since Rails 5.2 supports Ruby 2.2 and it doesn't support `RUBY_ENGINE_VERSION`, https://github.com/rails/rails/commit/f2201f1101547ef55df753c4a5b0529801889737) to get JRuby version in order to write it down to generated `.ruby-version` file.

### Actual Behavior

```
$ echo "jruby-9.2.0.0-SNAPSHOT" > .ruby-version
$ rbenv version
rbenv: version `jruby-9.2.0.0-SNAPSHOT' is not installed (set by /path/to/repo/.ruby-version)
 (set by /path/to/repo/.ruby-version)
```

```
$ echo "jruby-9.2.0.0-dev" > .ruby-version
$ rbenv version
jruby-9.2.0.0-dev (set by /path/to/repo/.ruby-version)
```

The problem is `RUBY_ENGINE_VERSION`, `JRUBY_VERSION` return `"9.2.0.0-SNAPSHOT"` instead of `"9.2.0.0-dev"` under JRuby dev version.

**Comment by bogdanvlviv** [comment]
Not sure, maybe it should be fixed in https://github.com/rbenv/ruby-build

**Comment by headius** [comment]
@bogdanvlviv After discussing this with @mkristian it seems like this should be fixed in ruby-build. We have not used the -dev suffix for quite some time, and all our development builds will have the -SNAPSHOT prefix standard to maven builds. ruby-build is mapping the -dev suffix to a download of the -SNAPSHOT build, which will always report RUBY_ENGINE_VERSION with the -SNAPSHOT suffix.

**Comment by bogdanvlviv** [comment]
@headius After investigation of this issue, I've realized that it is Rails issue.
We used to `JRUBY_VERSION` in order to get the version of JRuby, but `rbenv` and `rvm` have a different convention of naming the latest version/rc-releases of Ruby/JRuby. In order to get consistent with these Ruby version manages names of versions we should use `ENV["RBENV_VERSION"]`, and
`ENV["rvm_ruby_string"]`.
It should be fixed by https://github.com/rails/rails/pull/33152.
